kernel/src/thread/oops.rs

```diff
@@ -97,16 +97,27 @@ fn panic_handler(info: &core::panic::PanicInfo) -> ! {
         }
     }
 
+    let preempt_guard = disable_preempt();
+    let thread = Thread::current();
```

Comment:
This method involves heap allocations. I guess that we have to choose between more robust vs more informative panic handling. Which one do you prefer?

Of course. It is also possible to add new API to OSTD to get the current task without heap allocations.

Reply:
`Thread::current()` does not make new heap allocations. It only calls `Task::current`, and what `Task::current` do is increment the already allocated reference count and return an `Arc`. It does not de-allocate since the current task's reference count won't be decremented to zero is here. Printing does not allocate either since they are all tackled by `fmt`.

The only allocation in panic handling now is during the raise of the panic and expecting it to be caught (when `PANIC_ON_OOPS` is not set), which is not introduced by this PR :>

Reply:
> Of course. It is also possible to add new API to OSTD to get the current task without heap allocations.

Already heap free~

Reply:
Thanks for the explanation!